src/components/apps/toolbar/AppNavigation.js

```diff
@@ -101,41 +94,40 @@ function AppNavigation(props) {
     const [anchorEl, setAnchorEl] = useState(null);
 
     const appNavId = build(baseId, ids.APPS_NAVIGATION);
-    const allAppsCategory = getAllAppsCategory(constants.BROWSE_ALL_APPS);
+
+    const iconMap = new Map();
+    iconMap.set(constants.APPS_UNDER_DEV, <LockIcon />);
+    iconMap.set(constants.APPS_SHARED_WITH_ME, <FolderSharedIcon />);
+    iconMap.set(constants.FAV_APPS, <FavoriteIcon />);
+    iconMap.set(constants.MY_PUBLIC_APPS, <GroupWorkIcon />);
+    iconMap.set(constants.BROWSE_ALL_APPS, <AppsIcon />);
+    iconMap.set(constants.HPC, <StorageIcon />);
+
+    const allAppsCategory = useCallback(() => {
+        return {
+            name: constants.BROWSE_ALL_APPS,
+            id: constants.BROWSE_ALL_APPS_ID,
+        };
+    }, []);
 
     const preProcessData = useCallback(
         (data) => {
             const privateCat = data.categories.find(
-                (cat) => cat.system_id === "de"
+                (cat) => cat.system_id === systemId.de
             );
             const hpcCat = data.categories.find(
-                (cat) => cat.system_id === "agave"
+                (cat) => cat.system_id === systemId.agave
             );
 
-            privateCat.categories.forEach((category) => {
-                if (category.name === constants.APPS_UNDER_DEV) {
-                    category.icon = <LockIcon />;
-                } else if (
-                    category.name.toLowerCase() ===
-                    constants.SHARED_WITH_ME.toLowerCase()
-                ) {
-                    category.icon = <FolderSharedIcon />;
-                } else if (category.name === constants.FAV_APPS) {
-                    category.icon = <FavoriteIcon />;
-                } else if (category.name === constants.MY_PUBLIC_APPS) {
-                    category.icon = <GroupWorkIcon />;
-                }
-            });
-
+            let categoryList = privateCat.categories;
             if (hpcCat) {
-                hpcCat.icon = <StorageIcon />;
-                setCategories(privateCat.categories.concat(hpcCat));
-            } else {
-                setCategories(privateCat.categories);
+                categoryList = categoryList.concat(hpcCat);
             }
+            categoryList = categoryList.concat(allAppsCategory());
+            setCategories(categoryList);
             handleAppNavError(null);
         },
-        [setCategories, handleAppNavError]
+        [allAppsCategory, handleAppNavError]
```

Comment:
I'm not seeing a warning here, but I would think `setCategories` might still need to be in the useEffect's dependency array

Reply:
I will try adding it and see 🤷‍♂️